accounting/accounting.go

```diff
@@ -0,0 +1,188 @@
+		v = itof(value)
+		f = ftof(c.FactorAsFloat64())
+	)
+	f64, _ := newf().Quo(v, f).Float64()
+	return f64
+}
+
+// Exchange - Apply currency exchange rates to an amount.
+//
+// value - should always be given in the minor currency unit.
+// exchange - should always be given from the approved finance list.
+//
+// Rounding to the nearest even is a defined business rule.
+// Tills may round up to the nearest penny, but for reporting, the rule is
+// always to use banker's rounding.
+//
+// If unclear, see: // http://wiki.c2.com/?BankersRounding.
+func Exchange(c currency.Currency, value, exchange float64) float64 {
+	// this can happen for example when dealing
+	// with a GBP->GBP exchange, for example.
+	if exchange == 0 {
+		exchange = 1
+	}
+
+	var (
+		v = ftof(value)
+		f = ftof(c.FactorAsFloat64())
+		e = ftof(exchange)
+	)
+	// Here we divide the value, by it's minor currency
+	// unit factor, then divide it once more by the
+	// exchange rate.
+	// -> v / f / e
+	f64, _ := v.Quo(v, f).Quo(v, e).Float64()
+
+	// basic banker's rounding
+	// http://wiki.c2.com/?BankersRounding
+	return math.RoundToEven(f64*100) / 100
+}
+
+// RatNetAmount applies a VAT rate to a big.Rat value. This method returns a big.Float
+// so it's accuracy can be checked, and it's value applied with .Rat(some.field)
+func RatNetAmount(gross, rate *big.Rat) (*big.Float, error) {
+	// Here we go for octuple precision as we are dealing with rational numbers.
+	bf := func(rat *big.Rat) *big.Float {
+		return big.NewFloat(0).SetRat(rat).SetPrec(OctuplePrecision).SetMode(big.ToNearestEven)
+	}
+
+	v := bf(gross)
+	r := bf(rate)
+
+	// Guard against impossible (negative) tax rates.
+	switch r.Cmp(min) {
+	case -1:
+		return min, ErrSubZeroRate
+	case 0:
+		return v, nil
+	}
+	// Turn the rate into a divisor by making it superior to 1.
+	divisor := newf().Add(base, r)
+
+	// Here we divide the gross by it's vat:
+	// -> val / vat
+	// where vat is a gross superior to 1.
+	return newf().Quo(v, divisor), nil
+}
+
+// NetAmount derives the net amount before tax is applied using the given rate.
+func NetAmount(gross int64, rate float64) (int64, error) {
+	// Coerce the amount and rate into big floats to perform accurate calculations.
+	g := itof(gross)
+	r := ftof(rate)
+	// Guard against impossible (negative) tax rates.
+	switch r.Cmp(min) {
+	case -1:
+		return 0, ErrSubZeroRate
+	case 0:
+		return gross, nil
+	}
+	// Turn the rate into a divisor by making it superior to 1.
+	divisor := newf().Add(base, r)
+	// The net amount must be:
+	// amount / (rate + 1)
+	netFloat := newf().Quo(g, divisor)
+	// To avoid integer rounding errors we pass the float into a string first then cast to an integer.
+	netStr := netFloat.Text('f', 0)
+	net, _ := new(big.Int).SetString(netStr, 10)
+	// Derive the integer value of the net amount
+	return net.Int64(), nil
+}
+
+// TaxAmount returns the difference between the gross and the net amounts.
+func TaxAmount(gross, net int64) (int64, error) {
```

Comment:
If we're going to have a `RatNetAmount`, I think we should have a `RatTaxAmount`. Do you agree?

Reply:
`big.Rat` is really useful for precise divisions/operations on floating point numbers.
Here we can get away with just doing standard integer arithmetic, let's not complicate anything!